### PR TITLE
Fix support for EXTPATH on windows

### PR DESCRIPTION
**Pull request by steelbrain**
Related https://github.com/steelbrain/atom-linter/issues/129

### CODE REVIEW

File: src/helpers.js [hunk 1]
```diff
@@ -71,3 +70,23 @@ export async function getSpawnOptions(options: Exec$Options): Promise {
   }
   return spawnOptions
 }
+
+export function mergeAllPaths(env: Object): string {
+  let allOfIt = ''
+  for (const key in env) {
+    if (env.hasOwnProperty(key) && key.toUpperCase() === 'PATH') {
+      allOfIt += ';' + env[key]
```
Comment (Arcanemagus) [inline src/helpers.js]: This will result in the string starting with a `;` (blank entry).

File: src/helpers.js [hunk 2]
```diff
@@ -71,3 +70,23 @@ export async function getSpawnOptions(options: Exec$Options): Promise {
...
+    }
+  }
+  return allOfIt
+}
+
+export function mergeAllPathExts(env: Object): string {
+  let allOfIt = ''
+  for (const key in env) {
+    if (env.hasOwnProperty(key) && key.toUpperCase() === 'EXTPATH') {
+      allOfIt += ';' + env[key]
```
Comment (Arcanemagus) [inline src/helpers.js]: Same here

**Comment by Arcanemagus** [comment]
I'm not sure that was the issue, but utilizing that module isn't a bad idea I suppose.

**Comment by steelbrain** [comment]
Addressed the comments, and I think this is the issue because of joyent/node#2318

**Comment by Arcanemagus** [comment]
Is there something majorly different in how this handles PATHEXT from how it was done before? The problem builds worked just fine before this so I don't think the PATHEXT is the issue...

**Comment by steelbrain** [comment]
Well previously the `atom-linter`'s exec came from buffered-process which takes care of this by spawning every process from a spawned cmd and doing a lot of crazy escaping, builds started crashing because we hadn't taken care of it in this library

This PR aims to fix that issue

**Comment by Arcanemagus** [comment]
I'm still seeing this issue even with the changes in this branch (tested while reproducing https://github.com/AtomLinter/linter-bootlint/issues/48).

**Pull request merged**